source/retryable-writes/retryable-writes.rst

```diff
@@ -66,10 +66,50 @@ ClientSession
    name of this object MAY vary across drivers.
 
 Retryable Error
-   Any network exception or a write command response with an error indicating
-   that the node is no longer a primary (e.g. "not master" or "node is
-   recovering" errors). Such errors correspond to those discussed in the SDAM
-   spec's section on `Error Handling`_.
+   An error is considered retryable if it meets any of the following criteria:
+
+   - any network exception (e.g. socket timeout or error)
+
+   - a server error response with any the following codes:
+
+     .. list-table::
+       :header-rows: 1
+
+        * - Error Name
+          - Error Code
+        * - InterruptedAtShutdown
+          - 11600
+        * - InterruptedDueToReplStateChange
+          - 11602
+        * - NotMaster
+          - 10107
+        * - NotMasterNoSlaveOk
+          - 13435
+        * - NotMasterOrSecondary
+          - 13436
+        * - PrimarySteppedDown
+          - 189
+        * - ShutdownInProgress
+          - 91
+        * - WriteConcernFailed
+          - 64
+        * - HostNotFound
+          - 7
+        * - HostUnreachable
+          - 6
+        * - NetworkTimeout
+          - 89
+        * - SocketException
+          - 9001
+
+   - a server error response without an error code or one different from those
+     listed above, but with an error message containing the substring "not
+     master" or "node is recovering"
```

Comment:
One reason we should move away from this kind of error message checking in the future:
```
MongoDB Enterprise > db.runCommand({create:"not master"})
{ "ok" : 1 }
MongoDB Enterprise > db.runCommand({create:"not master"})
{
	"ok" : 0,
	"errmsg" : "a collection 'test.not master' already exists",
	"code" : 48,
	"codeName" : "NamespaceExists"
}
```

I don't think this causes any problems right now but it could in the future.

Reply:
Neat, good point. Should we make note of that somewhere?

Reply:
Another terrific catch, @ShaneHarvey. Could you open a SPEC ticket with that example and a proposal to stop parsing error messages as soon as we drop the newest server version that returns errors without codes? (I don't know which version that is: maybe 3.2 or 3.4.)

Reply:
Opened: https://jira.mongodb.org/browse/SPEC-1056